examples/Linux/README.md

```diff
@@ -0,0 +1,106 @@
+# wolfSentry lwIP Echo Test
+
+This is an extremely basic demo application that uses wolfSentry with the lwIP stack via PCAP on Docker to echo anything sent to it. wolfSentry will filter various types of traffic which can be tested using three Docker nodes the test generates.
+
+## Prerequisites
+
+This test is designed to be run on Linux or macOS, but should work on Windows as well. You need the following installed on your computer:
+
+* Docker - <https://docs.docker.com/get-docker/>
+* docker-compose - <https://docs.docker.com/compose/install/>
+
+## Running echo server
+
+The following command will build the test echo server and start this up along with three testing nodes:
+
+```sh
+sudo docker-compose -f docker-compose.yml up
+```
+
+You can follow the logs for the echo server using:
+
+```sh
+sudo docker-compose -f docker-compose.yml logs -f
+```
+
+It is recommended that you keep the logs following running whilst running the tests.
+
+## Testing
+
+### Accessing nodes
+
+There are three user test nodes to play with. They are named `wolfsentry-test_tester?_1` where `?` is 1, 2 or 3. To log into tester2 as an example:
+
+```sh
+sudo docker exec -it wolfsentry-test_tester2_1 /bin/sh
+```
+
+### Node details
+
+#### Echoserver
+
+* IP address: 172.20.20.3 (node) 127.20.20.5 (echo process)
+* MAC address: de:c0:de:01:02:03
+
+The echo test process runs from this node, it uses PCAP and lwIP to create a static IP of 127.20.20.5 for the actually test.
+
+#### Tester 1
+
+* IP address: 172.20.20.10
+* MAC address: de:c0:de:03:02:01
+
+The sentry test is configured to allow this node to ping the echoserver node, but not TCP connect.
+
+#### Tester 2
+
+* IP address: 172.20.20.20
+* MAC address: de:c0:de:03:02:02
+
+The sentry test is configured to block this node pinging the echoserver node, but allow TCP connect.
+
+#### Tester 3
+
+* IP address: 172.20.20.20
+* MAC address: de:c0:de:03:03:01
+
+The sentry test is configured to deny traffic from this MAC address.
```

Comment:
The server log correctly shows
```
echoserver_1  | MAC Sentry action return: 67395072
echoserver_1  | Sentry rejected MAC address
echoserver_1  | MAC Sentry action return: 67395072
echoserver_1  | Sentry rejected MAC address
echoserver_1  | MAC Sentry action return: 67395072
echoserver_1  | Sentry rejected MAC address
echoserver_1  | MAC Sentry action return: 67395072
echoserver_1  | Sentry rejected MAC address
echoserver_1  | MAC Sentry action return: 67395072
echoserver_1  | Sentry rejected MAC address
echoserver_1  | MAC Sentry action return: 67395072
echoserver_1  | Sentry rejected MAC address
```
and indeed the connection is rejected:
```
/ # nc -v 172.20.20.5 11111
nc: 172.20.20.5 (172.20.20.5:11111): Host is unreachable
```
However, ICMP is not rejected:
```
/ # ping 127.20.20.5
PING 127.20.20.5 (127.20.20.5): 56 data bytes
64 bytes from 127.20.20.5: seq=0 ttl=64 time=0.042 ms
```
and the server log is silent while these ICMP packets are bouncing around.

Reply:
I cannot reproduce this. What platform is your Docker sitting on? (I'm guessing ICMP isn't hitting LWIP).